src/libmongoc/src/mongoc/mongoc-client-side-encryption.c

```diff
@@ -2161,13 +2169,33 @@ mongoc_client_encryption_get_key (mongoc_client_encryption_t *client_encryption,
                               client_encryption->keyvault_coll)),
                         MONGOC_READ_CONCERN_LEVEL_MAJORITY) == 0);
 
-   /* If an error occurred, user should query cursor error. */
+   if (key_doc) {
+      memset (key_doc, 0, sizeof (*key_doc));
+   }
+
    cursor = mongoc_collection_find_with_opts (
       client_encryption->keyvault_coll, &filter, NULL, NULL);
 
+   ret = !mongoc_cursor_error (cursor, error);
+
+   if (ret && key_doc) {
+      const bson_t *bson = NULL;
+
+      if (!mongoc_cursor_next (cursor, &bson)) {
+         key_doc->value_type = BSON_TYPE_NULL;
+      } else {
+         bson_t *const doc = BCON_NEW ("v", BCON_DOCUMENT (bson));
+         bson_iter_t iter;
+         BSON_ASSERT (bson_iter_init_find (&iter, doc, "v"));
+         bson_value_copy (bson_iter_value (&iter), key_doc);
+         bson_destroy (doc);
```

Comment:
Note: @kevinAlbs also shared an alternative implementation in [Slack](https://mongodb.slack.com/archives/GMY98JS86/p1653499268370519?thread_ts=1653496710.212739&cid=GMY98JS86).

Reply:
I opted for this method as there are fewer opportunities for mistakes despite the unnecessary allocation.